server/api/actions.go

```diff
@@ -102,34 +102,22 @@ func isValidTrigger(trigger string) bool {
 	return false
 }
 
-func isValidAction(action string) bool {
-	if action == "" {
-		return true
-	}
-
-	for _, elem := range app.ValidActionTypes {
-		if action == string(elem) {
-			return true
-		}
-	}
-
-	return false
-}
-
 func parseGetChannelActionsOptions(query url.Values) (*app.GetChannelActionOptions, error) {
 	actionTypeStr := query.Get("action_type")
 	triggerTypeStr := query.Get("trigger_type")
 
-	if !isValidAction(actionTypeStr) {
-		return nil, fmt.Errorf("action_type %q not recognized; valid values are %v", actionTypeStr, app.ValidActionTypes)
+	var actionType app.ActionType
+	err := (&actionType).UnmarshalText([]byte(actionTypeStr))
```

Comment:
I feel like this would "read" better as:
```suggestion
	actionType, err := NewAtionTypeFromString(actionTypeStr)
```

Reply:
Like it. If we go this way, I would do directly `actionType, errr := NewActionType(s string)`

In the end, there is no other constructor, and seems clearer.